chain/network/src/types.rs

```diff
@@ -618,7 +617,7 @@ impl RawRoutedMessage {
     }
 }
 
-#[derive(BorshSerialize, Serialize, PartialEq, Eq, Clone, Debug)]
+#[derive(BorshSerialize, PartialEq, Eq, Clone, Debug)]
```

Comment:
*For other reviewers:*

@khorolets and I had a chat about where should `BlockReference` structure belong, and we decided to keep `BlockReference` structure in core-primitives, but remove `serde` serialization/deserialization from it since it was JSON-RPC-specific and could easily confuse people or introduce a backward-incompatible change. We tried to do so, and identified unused serialization/deserialization derives on near-network structures.

@mfornet Do we need serde in `near-network` at all? It feels unnecessary to me, but I see a few uses, which I believe could be either extracted from `near-network` or replaced with Debug display instead of using JSON serialization.

Reply:
@bowenwang1996 Ok, we just cleaned up the unused bits that were needed to create a clear separation of concerns, and won't do anything beyond that.

Reply:
What I meant is that this change will completely deprecate the rpc routing and we should be aware of that. I am fine doing with but would like to be sure that we are on the same page.

Reply:
I don't think rpc routing is relevant here. RPC routing is done through near-network, which uses borsh for serialization (RPC receives the data in JSON, deserializes it, view-client can still route the request through near-network, which will take the structure and serialize it with borsh, and thus JSON serialization is relevant only on the RPC side)

Reply:
Yeah good point. I cannot remember why we need `Serialize` here. Maybe it was simply a mistake